rkt: Get logs via syslog identifier

euank opened the pull request:
This change works around https://github.com/coreos/rkt/issues/2630
Without this change, logs cannot reliably be collected for containers
with short lifetimes.

With this change, logs cannot be collected on rkt versions v1.6.0 and
before.

I'd like to also bump the required rkt version, but I don't want to do that until there's a released version that can be pointed to (so the next rkt release).

I haven't added tests (which were missing) because this code will be removed if/when logs are retrieved via the API. I have run E2E tests with this merged in and verified the tests which previously failed no longer fail.

cc @yifan-gu

euank committed:
rkt: Get logs via syslog identifier

This change works around https://github.com/coreos/rkt/issues/2630
Without this change, logs cannot reliably be collected for containers
with short lifetimes.

With this change, logs cannot be collected on rkt versions v1.6.0 and
before.

yifan-gu commented:
Thanks for the fix. This would help fixes a bunch of rktnetes regression recently, right?

yifan-gu commented:
LGTM

yifan-gu commented:
Removing LGTM until rkt 1.7 is cut.

euank commented:
```
--- FAIL: TestUnschedulableNodes (20.53s)
	scheduler_test.go:166: Failed to observe reflected update for setting unschedulable=true: timed out waiting for the condition
```

@k8s-bot unit test this issue: #12312

euank committed:
rkt: Update minimum rkt version to 1.7.0

Also remove the redundant `appcVersion` check, that version should
already be captured in the rkt version

euank commented:
Bumped to requiring rkt `v1.7.0` now that that's released, also removed the appc version. PTAL @yifan-gu

spxtr commented:
@k8s-bot e2e test this issue: #23545
